framework/helpers/BaseJson.php

```diff
@@ -109,6 +109,31 @@ public static function decode($json, $asArray = true)
         return $decode;
     }
 
+    /**
+     * Validates the given data and returns a boolean indicating whether
+     * it is a valid JSON string or not.
+     *
+     * @param mixed $data data to validate.
+     * @return boolean `true` if given data is a valid JSON string, `false` otherwise.
+     *
+     * @since 2.0.14
+     */
+    public static function validate($data)
+    {
+        $isValid = false;
+
+        if (is_string($data) === true) {
+            json_decode($data);
+
+            $lastError = json_last_error();
+            if ($lastError === JSON_ERROR_NONE) {
```

Comment:
This info is valuable so I'd think twice before returning just true/false and losing it.

Reply:
http://www.beliefmedia.com/code/php-snippets/check-valid-json

Reply:
@samdark I refactored the method so that now it returns actual validation errors.

Reply:
@samdark another option is to return the error via the output parameter. In fact I like this option more.